cdap-watchdog/src/main/java/co/cask/cdap/logging/save/KafkaMessageCallback.java

```diff
@@ -100,11 +92,19 @@ public void onReceived(Iterator<FetchedMessage> messages) {
                   message.getTopicPartition().getTopic(),
                   message.getTopicPartition().getPartition());
       }
-
-      count++;
     }
 
-    if (count > 0) {
+    int count = events.size();
+    if (!events.isEmpty()) {
+      for (KafkaLogProcessor processor : kafkaLogProcessors) {
+        try {
+          processor.process(events.iterator());
```

Comment:
so iterator with same set of events will be passed to each process method of processor, and how would they work ? just trying to understand, when one of the processor finds the events aren't already processed in AbstractKafkaLogProcesssor and calls process, another processor can also do the same right ?